Geo3x3.php

```diff
@@ -23,14 +25,19 @@ public static function encode(float $lat, float $lng, int $level): string {
     }
     return $res;
   }
-  public static function decode(string $code): array {
+
+  /**
+   * @phpstan-return array{float, float, positive-int, float}
+   */
+  public static function decode(string $code): array
+  {
     $c = $code[0];
     $begin = 0;
     $flg = false;
-    if ($c == "-" || $c == "W") {
+    if (in_array($c, ['-', 'W'], true)) {
```

Comment:
Although it looks like more code, it compiles into a PHP opcode IN_ARRAY instruction instead of a function call, which saves execution time. (But this is not a bottleneck, it's just a micro-optimization)